typescript/src/services/redemptions/redemptions-service.ts

```diff
@@ -170,6 +184,94 @@ export class RedemptionsService {
     return { walletPublicKey, mainUtxo, redeemerOutputScript }
   }
 
+  /**
+   *
+   * @param bitcoinRedeemerAddress Bitcoin address redeemed BTC should be
+   *                               sent to. Only P2PKH, P2WPKH, P2SH, and P2WSH
+   *                               address types are supported.
+   * @param amount The amount to be redeemed with the precision of the tBTC
+   *                on-chain token contract.
+   * @param potentialCandidateWallets Array of wallets that can handle the
+   *                                  redemption request. The wallets must
+   *                                  be in the Live state.
+   * @returns Object containing:
+   *          - Bitcoin public key of the wallet asked to handle the redemption.
+   *           Presented in the compressed form (33 bytes long with 02 or 03 prefix).
+   *         - Wallet public key hash.
+   *         - Main UTXO of the wallet.
+   *         - Redeemer output script.
+   *
+   * @throws Throws an error if no valid redemption wallet exists for the given
+   *         input parameters.
+   */
+  protected async determineValidRedemptionWallet(
+    bitcoinRedeemerAddress: string,
+    amount: BigNumber,
+    potentialCandidateWallets: Array<SerializableWallet>
+  ): Promise<{
+    walletPublicKey: Hex
+    mainUtxo: BitcoinUtxo
+    redeemerOutputScript: Hex
+  }> {
```

Comment:
Nit: This could be a type
```suggestion
  ): Promise<RedemptionWallet | null> {
```

Reply:
Addressed on [refactor: add redemption wallet type](https://github.com/keep-network/tbtc-v2/pull/850/commits/0528fcd08b28a2436e507400903431a6271e66ab)